webpack/assets/javascripts/services/ChartService.js

```diff
@@ -30,15 +31,48 @@ const getChartConfig = ({
   const chartConfigForType = sizeConfig[config];
   const colors = getColors(data);
   const colorsSize = Object.keys(colors).length;
+  const dataExists = doDataExist(data);
+
+  const longNames = [];
+  const shortNames = [];
+
+  let dataWithShortNames = [];
+
+  if (dataExists) {
+    dataWithShortNames = data.map((val) => {
+      const item = Immutable.asMutable(val.slice());
+      longNames.push(item[0]);
+      item[0] = item[0].length > 30 ? `${val[0].substring(0, 10)}...` : item[0];
+      shortNames.push(item[0]);
+      return item;
+    });
+  }
 
   return {
     ...chartConfigForType,
     id,
     data: {
-      columns: doDataExist(data) ? data : [],
+      columns: dataExists ? dataWithShortNames : [],
       onclick,
       ...(colorsSize > 0 ? { colors } : {}),
     },
+    // eslint-disable-next-line
```

Comment:
why did you need to disable linting here?

Reply:
updated